### PR TITLE
feat(rule): add time-bound tare exception support for truck containers

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### 🧾 Summary

This PR adds support for time-bound tare exceptions in the `weighing`rule processor, allowing recyclers with legacy manual weighing systems to submit historical `TRUCK` container weighing events without tare values when they have an approved, non-expired exception.

### 🧩 Context

Some recyclers have legacy manual weighing systems that only captured net weight for `TRUCK` containers and did not record tare (empty container weight). To enable these recyclers to submit historical mass data for accreditation, we need to support exceptions that allow missing tare values under specific conditions.

This change unblocks historical data submission while maintaining data quality standards through:
- Time-bound exceptions (Valid Until date)
- Container type restrictions (TRUCK only)
- Robust date validation

**Test Scenarios Covered:**
1. ✅ TRUCK container + valid tare exception + missing Tare → **PASS** with exception message
2. ✅ TRUCK container + no exception + missing Tare → **FAIL** (existing behavior)
3. ✅ BIN container + tare exception + missing Tare → **FAIL** (exception doesn't apply to non-TRUCK)
4. ✅ TRUCK container + tare exception + Tare provided → **PASS** (normal validation)
5. ✅ TRUCK container + expired exception + missing Tare → **FAIL** (exception expired)
6. ✅ TRUCK container + invalid date format exception + missing Tare → **FAIL** (invalid date)

**Expected Results:**
- All 68 tests pass
- 100% code coverage maintained
- No regressions in existing functionality

### 🧠 Considerations for Review

**Key Implementation Details:**
- **Date Validation Logic**: Exception is valid when `today < Valid Until`. Uses date-fns for robust date parsing and comparison
- **Container Type Restriction**: Exception only applies to `TRUCK` containers (not BIN, CAGE, etc.)
- **Exception Extraction**: Tare exceptions are extracted from recycler accreditation document, not the weighing event itself
- **Message Enhancement**: Pass messages now include the Valid Until date when present for transparency

**Architectural Decisions:**
- Comparison against current date (not document creation date) ensures exceptions are validated at processing time
- Invalid date formats are treated as invalid exceptions (fail-safe approach)
- Exception without Valid Until field is considered permanently valid (backwards compatible)

**Trade-offs:**
- Single return statement in `isExceptionValid()` for better code quality (SonarQube compliance)
- Verbose test cases ensure comprehensive edge case coverage

---

- [x] **I, the PR author, confirm that this PR works as expected and does not break any service. I also confirm that I applied best practices and improved the codebase quality.**

## Summary by CodeRabbit

* **New Features**
  * Tare may be omitted when an approved tare exception is present; exceptions can include an optional "Valid Until" date which is validated and may expire.
  * Container-capacity exceptions are now validated by date as well, and validation logic uses exception validity to determine pass behavior.
  * When a valid tare exception applies, the pass message explicitly notes that omission of the tare was permitted.

* **Tests**
  * Expanded coverage for tare exceptions (valid, expired, invalid dates), various container types, tare present/absent, and container-capacity scenarios.

### CODE REVIEW

File: libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.processor.ts [hunk 1]
```diff
@@ -127,6 +127,14 @@ export class WeighingProcessor extends RuleDataProcessor {
       passMessage = PASSED_RESULT_COMMENTS.PASSED_WITH_EXCEPTION(passMessage);
     }
 
+    if (
+      !isNil(weighingValues.tareException) &&
```
Comment (cris-santos) [inline libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.processor.ts]: Just to double-check, only checking that's not nil is enough?

File: libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.helpers.ts [hunk 2]
```diff
@@ -105,6 +107,28 @@ export const getMandatoryFieldExceptionFromAccreditationDocument = (
...
+  }
+
+  const validUntilDate = parseISO(validUntil);
+
+  if (!isValid(validUntilDate)) {
+    return false;
+  }
+
+  return !isAfter(new Date(), validUntilDate);
+};
```
Comment (cris-santos) [inline libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.helpers.ts:178]: Just to double-check, don't we need to check the exception content? Just being defined is enough?

**Review by cris-santos — DISMISSED** [review]

**Review by GLGuilherme — DISMISSED** [review]

File: libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.processor.ts (hunk 1 above)
Comment (AMarcosCastelo) [reply libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.processor.ts]: Good point! https://github.com/carrot-foundation/methodology-rules/pull/312/commits/72342a10d42f21eeaccf084bb900b87d5780ee74

File: libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.helpers.ts (hunk 2 above)
Comment (AMarcosCastelo) [reply libs/methodologies/bold/rule-processors/mass-id/weighing/src/weighing.helpers.ts:178]: Good point! https://github.com/carrot-foundation/methodology-rules/pull/312/commits/72342a10d42f21eeaccf084bb900b87d5780ee74

**Review by andtankian — APPROVED** [review]

**Review by GLGuilherme — APPROVED** [review]

**Review by cris-santos — APPROVED** [review]
